### PR TITLE
Target 10.9.z

### PR DESCRIPTION
**Pull request by jumoog**

### CODE REVIEW

**Comment by AbandonedCart** [comment]
.net 8 breaks support for 32 bit. They're going to get a lot of backlash for it.

**Comment by jumoog** [comment]
but https://dotnet.microsoft.com/en-us/download/dotnet/8.0 has 32bit support?

**Comment by AbandonedCart** [comment]
It may have just been Mac then. I had heard it was 32 in general. I know Jellyfin decided to jump their minimum OS from 10 to 12 for Mac because of it.

**Comment by AbandonedCart** [comment]
I think it might be good to have a branch that's updated and one that isn't. I'm sure there will be stuff in the new update that requires changes, but then we don't throw out people stuck on 10.8

File: ConfusedPolarBear.Plugin.IntroSkipper/ConfusedPolarBear.Plugin.IntroSkipper.csproj [hunk 1]
```diff
@@ -11,8 +11,8 @@
     <CodeAnalysisRuleSet>../jellyfin.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Jellyfin.Controller" Version="10.9.*" />
-    <PackageReference Include="Jellyfin.Model" Version="10.9.*" />
+    <PackageReference Include="Jellyfin.Controller" Version="10.8.13" />
```
Comment (jumoog) [inline ConfusedPolarBear.Plugin.IntroSkipper/ConfusedPolarBear.Plugin.IntroSkipper.csproj:14]: must be 10.9.* we need to wait for nuget packages

Comment (AbandonedCart) [reply ConfusedPolarBear.Plugin.IntroSkipper/ConfusedPolarBear.Plugin.IntroSkipper.csproj:14]: Yeah, but it's not available yet. This was just to test building.

**Comment by AbandonedCart** [comment]
Did you create a secret named "token" or are you trying to use the github one?

**Comment by jumoog** [comment]
> Nuget is the release feed, github is the prerelease feed

Using Github as a NuGet source is absolute pain. I dont waste any more of my time trying to get Github as a source to work.

**Comment by AbandonedCart** [comment]
That's usually how it goes with that stuff.

**Comment by jumoog** [comment]
I'm so close to archiving this repository. I have no more motivation.

**Comment by AbandonedCart** [comment]
I got nuget working anyway

**Comment by AbandonedCart** [comment]
Looks like CodeQL for csharp isn't going to be able to check anything, since there's no config file

**Comment by jumoog** [comment]
```[2024-03-28 17:03:15.505 +01:00] [ERR] [40] Emby.Server.Implementations.Updates.InstallationManager: Package installation failed
System.IO.DirectoryNotFoundException: Could not find a part of the path '/config/plugins/Intro Skipper_0.1.16.4/logo.png'.
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirError)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, UnixFileMode openPermissions, Int64& fileLength, UnixFileMode& filePermissions, Boolean failForSymlink, Boolean& wasSymlink, Func`4 createOpenException)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at MediaBrowser.Model.IO.AsyncFile.OpenWrite(String path)
   at Emby.Server.Implementations.Plugins.PluginManager.PopulateManifest(PackageInfo packageInfo, Version version, String path, PluginStatus status)
   at Emby.Server.Implementations.Updates.InstallationManager.PerformPackageInstallation(InstallationInfo package, PluginStatus status, CancellationToken cancellationToken)
   at Emby.Server.Implementations.Updates.InstallationManager.PerformPackageInstallation(InstallationInfo package, PluginStatus status, CancellationToken cancellationToken)
   at Emby.Server.Implementations.Updates.InstallationManager.InstallPackageInternal(InstallationInfo package, CancellationToken cancellationToken)
   at Emby.Server.Implementations.Updates.InstallationManager.InstallPackage(InstallationInfo package, CancellationToken cancellationToken)
[2024-03-28 17:03:15.514 +01:00] [ERR] [40] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request: "Could not find a part of the path '/config/plugins/Intro Skipper_0.1.16.4/logo.png'". URL "POST" "/Packages/Installed/Intro%20Skipper".```

**Comment by jumoog** [comment]
oh the zip is empty :(

**Comment by jumoog** [comment]
with files it works

**Comment by jumoog** [comment]
current problem:
```
[2024-03-28 23:48:18.715 +01:00] [ERR] [21] Emby.Server.Implementations.ScheduledTasks.TaskManager: Error executing Scheduled Task
System.AggregateException: One or more errors occurred. (A non-collectible assembly may not reference a collectible assembly.)
 ---> System.NotSupportedException: A non-collectible assembly may not reference a collectible assembly.
   at System.Reflection.Emit.RuntimeModuleBuilder.<GetTypeRef>g____PInvoke|13_0(QCallModule __module_native, UInt16* __strFullName_native, QCallModule __refedModule_native, Int32 __tkResolution_native)
   at System.Reflection.Emit.RuntimeModuleBuilder.GetTypeTokenWorkerNoLock(Type type, Boolean getGenericDefinition)
   at System.Reflection.Emit.RuntimeModuleBuilder.GetTypeTokenInternal(Type type, Boolean getGenericDefinition)
   at System.Reflection.Emit.SignatureHelper.AddOneArgTypeHelperWorker(Type clsArgument, Boolean lastWasGenericInst)
   at System.Reflection.Emit.SignatureHelper.AddOneArgTypeHelperWorker(Type clsArgument, Boolean lastWasGenericInst)
   at System.Reflection.Emit.RuntimeILGenerator.DeclareLocal(Type localType, Boolean pinned)
   at System.Xml.Serialization.CodeGenerator.DeclareLocal(Type type, String name)
   at System.Xml.Serialization.CodeGenerator.DeclareOrGetLocal(Type type, String name)
   at System.Xml.Serialization.ReflectionAwareILGen.WriteArrayLocalDecl(String typeName, String variableName, SourceInfo initValue, TypeDesc arrayTypeDesc)
   at System.Xml.Serialization.XmlSerializationWriterILGen.WriteElement(SourceInfo source, ElementAccessor element, String arrayName, Boolean writeAccessor)
   at System.Xml.Serialization.XmlSerializationWriterILGen.WriteElements(SourceInfo source, String enumSource, ElementAccessor[] elements, TextAccessor text, ChoiceIdentifierAccessor choice, String arrayName, Boolean writeAccessors, Boolean isNullable)
   at System.Xml.Serialization.XmlSerializationWriterILGen.WriteMember(SourceInfo source, String choiceSource, ElementAccessor[] elements, TextAccessor text, ChoiceIdentifierAccessor choice, TypeDesc memberTypeDesc, Boolean writeAccessors)
   at System.Xml.Serialization.XmlSerializationWriterILGen.GenerateTypeElement(XmlTypeMapping xmlTypeMapping)
   at System.Xml.Serialization.TempAssembly.GenerateRefEmitAssembly(XmlMapping[] xmlMappings, Type[] types)
   at System.Xml.Serialization.TempAssembly..ctor(XmlMapping[] xmlMappings, Type[] types, String defaultNamespace, String location)
   at System.Xml.Serialization.XmlSerializer.GenerateTempAssembly(XmlMapping xmlMapping, Type type, String defaultNamespace, String location)
   at System.Xml.Serialization.XmlSerializer..ctor(Type type, String defaultNamespace)
   at Emby.Server.Implementations.Serialization.MyXmlSerializer.<>c.<GetSerializer>b__1_0(String _, Type t)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd[TArg](TKey key, Func`3 valueFactory, TArg factoryArgument)
   at Emby.Server.Implementations.Serialization.MyXmlSerializer.GetSerializer(Type type)
   at Emby.Server.Implementations.Serialization.MyXmlSerializer.SerializeToWriter(Object obj, XmlWriter writer)
   at Emby.Server.Implementations.Serialization.MyXmlSerializer.SerializeToStream(Object obj, Stream stream)
   at Emby.Server.Implementations.Serialization.MyXmlSerializer.SerializeToFile(Object obj, String file)
   at ConfusedPolarBear.Plugin.IntroSkipper.Plugin.SaveTimestamps()
   at ConfusedPolarBear.Plugin.IntroSkipper.Plugin.UpdateTimestamps(Dictionary`2 newTimestamps, AnalysisMode mode)
   at ConfusedPolarBear.Plugin.IntroSkipper.ChapterAnalyzer.AnalyzeMediaFiles(ReadOnlyCollection`1 analysisQueue, AnalysisMode mode, CancellationToken cancellationToken)
   at ConfusedPolarBear.Plugin.IntroSkipper.BaseItemAnalyzerTask.AnalyzeItems(ReadOnlyCollection`1 items, CancellationToken cancellationToken)
   at ConfusedPolarBear.Plugin.IntroSkipper.BaseItemAnalyzerTask.<>c__DisplayClass5_0.<AnalyzeItems>b__0(KeyValuePair`2 season)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass43_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.<>c__DisplayClass43_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
   at System.Threading.Tasks.TaskReplicator.Replica.Execute()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.TaskReplicator.Run[TState](ReplicatableUserAction`1 action, ParallelOptions options, Boolean stopOnFirstFailure)
   at System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body)
   at ConfusedPolarBear.Plugin.IntroSkipper.BaseItemAnalyzerTask.AnalyzeItems(IProgress`1 progress, CancellationToken cancellationToken)
   at ConfusedPolarBear.Plugin.IntroSkipper.DetectCreditsTask.ExecuteAsync(IProgress`1 progress, CancellationToken cancellationToken)
   at Emby.Server.Implementations.ScheduledTasks.ScheduledTaskWorker.ExecuteInternal(TaskOptions options)
[2024-03-28 23:48:18.718 +01:00] [INF] [21] Emby.Server.Implementations.ScheduledTasks.TaskManager: "Detect Credits" Failed after 0 minute(s) and 0 seconds
```

**Comment by AbandonedCart** [comment]
https://github.com/godotengine/godot/issues/66060

**Comment by jumoog** [comment]
using DataContract solves `A non-collectible assembly may not reference a collectible assembly.`
The only think left is autoskip.

**Comment by jumoog** [comment]
autoskip works now

**Pull request closed without merging**